ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/util/BatchWrite.java

```diff
@@ -176,7 +178,13 @@ private void recordBatchMetrics(long bytes, int messages, long delayMillis) {
 
     // wait for full then synchronize and close
     private final CompletableFuture<BatchResultT> result = full
-        .thenComposeAsync(this::synchronousClose, executor);
+        .thenComposeAsync(this::synchronousClose, executor).exceptionally(exception -> {
+          // with one input rethrow exception, otherwise use LoggedException to only log it once
+          throw this.size == 1 ? (RuntimeException) exception
```

Comment:
preserving the original exception for single-message batches prevents logging the exception here when it may still be caught and fixed, e.g. when the mixed output attempts to stream very large messages and then redirects them to file loads.